codebase2/codebase-sqlite/U/Codebase/Sqlite/NameLookups.hs

```diff
@@ -0,0 +1,64 @@
+-- | Types related to working with NameLookups.
+-- We define these low-level types rather than use Path's because we don't have
+-- access to those domain types given the package dependency tree.
+module U.Codebase.Sqlite.NameLookups
+  ( ReversedName (..),
+    ReversedPath (..),
+    PathSegments (..),
+    NamespaceText,
+    reversedNameToNamespaceText,
+    pathSegmentsToText,
+    textToPathSegments,
+  )
+where
+
+import Data.List.NonEmpty (NonEmpty)
+import Data.Text qualified as Text
+import Unison.Prelude
+
+newtype ReversedName = ReversedName (NonEmpty Text)
```

Comment:
The sqlite layer is too low in the dependency stack to depend on domain types like `Path`, but using bare haskell types or aliases proved too risky (a reversed name and a forwards name would unify)

So I added some newtypes.